index.js

```diff
@@ -114,8 +110,9 @@ function build (schema, options) {
         schemaKey = key + externalSchema.$id // relative URI
       }
 
-      if (ajvInstance.getSchema(schemaKey) === undefined) {
+      if (refResolver.getSchema(schemaKey) === undefined) {
```

Comment:
I would try to check the ajv's readonly properties.
This does not trigger the compiling and we could avoid the deep/equal logic

```
(x.schemas[schemaKey] || x.refs[schemaKey]) === undefined
```

```
const Ajv = require('ajv')

const x = new Ajv()

const externalSchema = {
  $id: 'root',
  definitions: {
    subschema: {
      $id: 'subschema',
      definitions: {
        anchorSchema: {
          $id: '#anchor',
          type: 'string'
        }
      }
    }
  }
}

x.addSchema(externalSchema)

{
  const id = 'subschema'
  console.log(x.schemas[id] || x.refs[id])
}

{
  const id = 'root'
  console.log(x.schemas[id] || x.refs[id])
}

{
  const id = 'subschema#anchor'
  console.log(x.schemas[id] || x.refs[id])
}
```

Reply:
The only concern I have is that refs and schemas is not a part of public API and can be changed without major release.

Reply:
Isn't it public?
https://github.com/ajv-validator/ajv/blob/master/lib/core.ts#L281

They would have declared as `private` instead